packages/style-engine/CONTRIBUTING.md

```diff
@@ -0,0 +1,40 @@
+# Contributing
+
+This document contains information you might need to know when extending or debugging Style Engine code.
+
+## Workflow and build tooling
+
+The Style Engine PHP and Javascript (JS) files exist inside the `style-engine` package. 
+
+In order to use the Style Engine in the Block Editor, these files must be compiled (in the case of JS) and copied to the build folder.
+
+When running the `npm run dev` script for example, webpack watches out for changes and will recompile/copy files as necessary if any changes are detected according to the rules in  [packages webpack config](https://github.com/WordPress/gutenberg/tree/HEAD/tools/webpack/packages.js).
+
+No other configuration is required for JS: webpack will compile and export the Style Engine code as it does with all dependencies listed in [package.json](https://github.com/WordPress/gutenberg/tree/HEAD/package.json).
+
+The PHP files for packages, however, have a couple of extra steps during the build process:
+
+1. Functions with the `wp_` prefix are replaced with `gutenberg_`. So, for example, `wp_some_function` becomes `gutenberg_some_function` in the build directory. The reason for this is so that the Block Editor can call Style Engine functions that may have evolved since, or have not yet been included in, any WordPress release.
+2. For the same reasons, classes are given a `_Gutenberg` suffix: `WP_Style_Engine` becomes `WP_Style_Engin_Gutenberg`. The [packages webpack config](https://github.com/WordPress/gutenberg/tree/HEAD/tools/webpack/packages.js) contains a static list of PHP classes (`bundledPackagesPhpConfig`) that have to be copied and renamed during build. If you create a new PHP class in the Style Engine package, you should add your classname to the `replaceClasses` array.
+
+Remember: all PHP functions and methods inside the Style Engine package should use `wp_/WP_` prefixes. 
+
+When updating existing PHP functions or methods, it's important to check the Block Editor codebase for calls to the equivalent `wp_` functions or classes as they may have to be updated to refer to `gutenberg_` or `_Gutenberg` in order for the updates to take effect.
```

Comment:
Should we also add a note about backporting too? (When the time comes to migrate changes from Gutenberg -> core, remove the prefixes / suffixes) Or is that kind of separate to the style engine package? I.e. for casual contributors, they'll likely not need to think about this 🤔

Reply:
I did start writing a section for backporting but it got a bit gnarly and deep since it's a niche subject. So I killed it for now according to the "When in doubt, chuck it out" philosophy. 😄 

I agree that we could write a minor note about it. Not sure where though. Maybe as part of a broader backport documentation.

Reply:
> So I killed it for now according to the "When in doubt, chuck it out" philosophy.

Sounds good, it's easy for us to add more detail further down the track if need be!